.github/workflows/ufm_log_analyzer_ci_workflow.yml

```diff
@@ -53,3 +53,29 @@ jobs:
         pip install ruff==0.7.3
 
         ruff format --diff --check src/loganalyze
+
+  pytest:
+    runs-on: ubuntu-22.04
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.9
+
+    - name: Run Pytest
+      run: |
+        SCRIPT_DIR="plugins/ufm_log_analyzer_plugin"
+        # Set PYTHONPATH to include src directory and two levels up for utils
+        PYTHONPATH="$(realpath $SCRIPT_DIR/src):$(realpath $SCRIPT_DIR/../../)"
+        export PYTHONPATH
+
+        cd $SCRIPT_DIR
+
+        pip install -r src/loganalyze/requirements.txt
+        pip install pytest
+
+        pytest
```

Comment:
Pytest should only run on the unit_tests directory:
1. There is no reason to let it scan the whole project (waist of time).
2. In the future we might add integration tests, they are also using the pytest frame work and we should separate the logics run

Reply:
done